### PR TITLE
Update kafka node

### PR DESCRIPTION
**Pull request by bkim54**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x]  Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/SOHU-Co/kafka-node>
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Review by sfrooster — APPROVED** [review]

**Comment by sfrooster** [comment]
@weswigham - it's been over 20 hours since this was merged (which I've heard is the upper interval between types-publisher runs) and the npm still hasn't been published.  Any insight?

File: types/kafka-node/index.d.ts [hunk 1]
```diff
@@ -6,26 +6,27 @@
...
-    constructor(client: Client, options?: ProducerOptions, customPartitioner?: any);
+    constructor(client: Client, options?: ProducerOptions, customPartitioner?: CustomPartitioner);
     on(eventName: "ready", cb: () => any): void;
     on(eventName: "error", cb: (error: any) => any): void;
     send(payloads: ProduceRequest[], cb: (error: any, data: any) => any): void;
-    createTopics(topics: string[], async: true, cb: (error?: any, data?: any) => any): void;
-    createTopics(topics: string[], async: false, cb: () => any): void;
-    createTopics(topics: string[], cb: (error?: any, data?: any) => any): void;
-    close(): void;
+    createTopics(topics: string[], async: boolean, cb: (error: any, data: any) => any): void;
```
Comment (amiram) [inline types/kafka-node/index.d.ts]: This method was intentionally split to true and false. When you pass false, you'll always get a callback without any arguments. When you pass true, you may get an error and/or data. Look here https://github.com/SOHU-Co/kafka-node/blob/master/lib/client.js#L363

File: types/kafka-node/index.d.ts [hunk 2]
```diff
@@ -6,26 +6,27 @@
...
+    createTopics(topics: string[], cb: (error: any, data: any) => any): void;
```
Comment (amiram) [inline types/kafka-node/index.d.ts]: I think error and data should stay optional. You might get undefined for them.

File: types/kafka-node/index.d.ts (hunk 1 above)
Comment (sfrooster) [inline types/kafka-node/index.d.ts]: ```(error: any, data: any) => any)```
is the same as
```(error?: any, data?: any) => any)```
(those "?"s are useless as ```any``` covers ```undefined```)
and it also therefore covers
```() => any)```

So this single declaration seems equivalent to (and less "wordy" than) the three it replaced.

**Pull request merged**